src/handleSubmit.js

```diff
@@ -56,33 +56,38 @@ const handleSubmit = (
       }
       if (isPromise(result)) {
         startSubmit()
-        return result.then(
-          submitResult => {
-            stopSubmit()
-            setSubmitSucceeded()
-            if (onSubmitSuccess) {
-              onSubmitSuccess(submitResult, dispatch, props)
+        return result
+          .then(submitResult => {
+            try {
```

Comment:
What is the purpose of this `try` block if you're just rethrowing the exception? Seems extraneous, but maybe I'm missing something.